src/Integration/MefServices/VsSessionHost.cs

```diff
@@ -336,8 +336,24 @@ private ILocalService GetConfigurationProvider()
 
             var bindingSerializer = new SolutionBindingSerializer(sccFileSystem, bindingFileLoader, credentialsLoader);
 
-            return new ConfigurationProvider(legacyConfigPathProvider, connectedModeConfigPathProvider, bindingSerializer,
-                legacyConfigFolderItemAdder);
+            return new ConfigurationPersister(legacyConfigPathProvider, connectedModeConfigPathProvider, bindingSerializer, legacyConfigFolderItemAdder);
+        }
+
+        private ILocalService GetConfigurationProvider()
```

Comment:
GetConfigurationProvider and GetConfigurationPersister are identical as the moment. This copy-paste will go away in future PRs, as I split more dependencies.